sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosItemRequestOptions.java

```diff
@@ -459,6 +499,30 @@ public Map<String, String> getHeader(CosmosItemRequestOptions cosmosItemRequestO
                 public CosmosDiagnosticsThresholds getDiagnosticsThresholds(CosmosItemRequestOptions cosmosItemRequestOptions) {
                     return cosmosItemRequestOptions.thresholds;
                 }
+
+                @Override
+                public boolean calculateAndGetEffectiveNonIdempotentRetriesEnabled(
+                    CosmosItemRequestOptions cosmosItemRequestOptions,
+                    Boolean clientDefault,
+                    boolean operationDefault) {
+
+                    if (cosmosItemRequestOptions.getNonIdempotentWriteRetriesEnabled() != null) {
+                        return cosmosItemRequestOptions.getNonIdempotentWriteRetriesEnabled();
+                    }
+
+                    if (!operationDefault) {
```

Comment:
is the operationDefault used to differentiate the default behavior between patch vs other create operations?

Reply:
Yes - patch and bulk I'll disable it